clients/src/main/java/org/apache/kafka/common/metrics/Metrics.java

```diff
@@ -192,9 +193,13 @@ public Metrics(MetricConfig defaultConfig, List<MetricsReporter> reporters, Time
      * @param tags        additional key/value attributes of the metric
      */
     public MetricName metricName(String name, String group, String description, Map<String, String> tags) {
-        Map<String, String> combinedTag = new LinkedHashMap<>(config.tags());
-        combinedTag.putAll(tags);
-        return new MetricName(name, group, description, combinedTag);
+        Map<String, String> combinedTag = new LinkedHashMap<>();
+        BiConsumer<String, String> combine = (k, v) -> {
+            combinedTag.put(k.intern(), v.intern());
+        };
+        config.tags().forEach(combine);
+        tags.forEach(combine);
+        return new MetricName(name.intern(), group.intern(), description, combinedTag);
```

Comment:
Likewise with interning name and group as they create the full metric name later.